src/components/vega-wallet/new-wallet-available.tsx

```diff
@@ -0,0 +1,56 @@
+import "./new-wallet-available.scss";
+
+import * as Sentry from "@sentry/react";
+import React from "react";
+import { useTranslation } from "react-i18next";
+
+import { Links } from "../../config/links";
+import { vegaWalletService } from "../../lib/vega-wallet/vega-wallet-service";
+
+export const NewWalletAvailable = () => {
+  const { t } = useTranslation();
+  const [newVersionAvailable, setNewVersionAvailable] = React.useState<
+    string | null
+  >(null);
+  const [latestVersionLocation, setLatestVersionLocation] =
+    React.useState<string>();
+
+  React.useEffect(() => {
+    const getWalletLatest = async () => {
+      try {
+        const res = await fetch(Links.WALLET_RELEASES_HISTORY, {});
+        const json = await res.json();
+
+        // find the first non-prelease
+        for (let walletVerison of json) {
+          if (!walletVerison.prerelease) {
+            const location = Links.WALLET_RELEASES_LATEST + walletVerison.name;
+            const latestVersion = walletVerison.name;
+            const [, currentVersion] = await vegaWalletService.getVersion();
```

Comment:
Can we get the wallet version once outside of the loop?

Reply:
it's not needed outside the loop.

Reply:
For each iteration of the loop `vegaWalletService.getVersion()` will return the same thing every time so I think its unnecessary to make a network request to the wallet every time.

Something like the below would be better 🕊️ 
```
const [, currentVersion] = await vegaWalletService.getVersion();

for (let walletVersion of json) {
  ...
  if (latestVersion !== currentVersion) {
    ... // setstate
  }
}
```

Reply:
Oh, I see, I thought you were pointing out that `latestVersion` was not available once the loop ended.